rmw_connext_cpp/bin/apply-patch.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# originated from the stackoverflow post:
+# https://stackoverflow.com/a/40967337
```

Comment:
@dirk-thomas do you think this note is going to be ok as is?

Reply:
Well, you can't just put OSRF copyright and Apachae license on the code copied from Stackoverflow. I guess you need to "reimplement the function if we can't find any API which does it for us.

Reply:
Apparently the code is public domain: https://gist.github.com/noporpoise/16e731849eb1231e86d78f9dfeca3abc#file-unifieddiff-py-L3

Reply:
Which means that IMO it's fine to license it as Apache 2.0.
It may be worth putting the 2016 copyright to the original author that's listed in the file ("Isaac Turner") to give credit even if it's public domain

Reply:
The copyright must be put here, and better attribution (maybe the GitHub username), also the public domain license should be mentioned here.

I'd recommend:

```
# Based on `unifieddiff.py` by @noporpoise which is licensed as `Public domain (CC0)`
# see: https://gist.github.com/noporpoise/16e731849eb1231e86d78f9dfeca3abc
```